src/compiler/checker.ts

```diff
@@ -13119,12 +13119,23 @@ namespace ts {
 
         // Return the contextual type for a given expression node. During overload resolution, a contextual type may temporarily
         // be "pushed" onto a node using the contextualType property.
-        function getApparentTypeOfContextualType(node: Expression): Type {
-            const type = getContextualType(node);
+        function getApparentTypeOfContextualType(node: Expression, checkMode?: CheckMode): Type {
+            const type = getContextualType(node, checkMode);
             return type && getApparentType(type);
         }
 
         /**
+         * This must be called when a contextual type function is attempting to dig into a type to contextually type a subexpression
+         * Otherwise, contextual typing stops when a type argument is encountered, and `any` is ued instead.
+         */
+        function getInstantiatedContextualType(node: Expression, checkMode?: CheckMode): Type {
+            if (checkMode === CheckMode.Inferential && node.contextualType && node.contextualMapper) {
+                return instantiateType(node.contextualType, node.contextualMapper);

```

Comment:
don't you still need to get an apparent type in case the result is still a type parameter?